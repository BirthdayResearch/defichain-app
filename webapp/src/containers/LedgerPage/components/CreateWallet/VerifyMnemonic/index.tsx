import React, { useEffect, useState } from 'react';
import { Helmet } from 'react-helmet';
import { I18n } from 'react-redux-i18n';
import { NavLink } from 'react-router-dom';
import classnames from 'classnames';
import { MdArrowBack, MdErrorOutline } from 'react-icons/md';
import { Row, Col, Button, Card } from 'reactstrap';
import { checkElementsInArray } from '../../../../../utils/utility';

import { WALLET_BASE_PATH, WALLET_PAGE_PATH } from '../../../../../constants';

import styles from '../CreateWallet.module.scss';
import { createWalletRequest, resetCreateWalletError } from '../../../reducer';
import { connect } from 'react-redux';
import WalletLoadingFooter from '../../../../../components/WalletLoadingFooter';

interface VerifyMnemonicProps {
  mnemonicObj: any;
  finalMixObj: any;
  mnemonicCode: string;
  isWalletTabActive: boolean;
  isWalletCreating: boolean;
  isErrorCreatingWallet: string;
  history: any;
  setIsWalletTabActive: (isWalletTabActive: boolean) => void;
  createWallet: (mnemonicCode: string, history: any) => void;
  resetCreateWalletError: () => void;
}

const VerifyMnemonic: React.FunctionComponent<VerifyMnemonicProps> = (
  props: VerifyMnemonicProps
) => {
  const [selectedWords, setSelectedWords] = useState<any>([]);
  const [mnemonicCheck, setMnemonicCheck] = useState(false);

  const {
    finalMixObj,
    isWalletTabActive,
    setIsWalletTabActive,
    mnemonicObj,
    history,
    isWalletCreating,
    isErrorCreatingWallet,
    mnemonicCode,
    createWallet,
    resetCreateWalletError,
  } = props;

  const handleSelect = (Obj) => {
    const tempArray = [...selectedWords, Obj];
    setSelectedWords(tempArray);
    const check = checkElementsInArray(tempArray, mnemonicObj);
    setMnemonicCheck(check);
  };

  const handleUnselect = (Obj) => {
    const filteredArray = selectedWords.filter((word) => Obj.key !== word.key);
    setSelectedWords(filteredArray);
  };

  useEffect(() => {
    props.resetCreateWalletError();
  }, []);

  return (
    <>
      <Helmet>
        <title>{I18n.t('containers.wallet.verifyMnemonicPage.title')}</title>
      </Helmet>
      <header className='header-bar'>
        <Button
          to={WALLET_BASE_PATH}
          tag={NavLink}
          color='link'
          className='header-bar-back'
        >
          <MdArrowBack />
          <span className='d-lg-inline'>
            {I18n.t('containers.wallet.verifyMnemonicPage.back')}
          </span>
        </Button>
        <h1 className={classnames({ 'd-none': false })}>
          {I18n.t('containers.wallet.verifyMnemonicPage.verifyMnemonicSeed')}
        </h1>
      </header>
      <div className='content'>
        <section>
          <p>
            {I18n.t(
              'containers.wallet.verifyMnemonicPage.verifyMnemonicGuideline'
            )}
          </p>
          <Row className='mb-3'>
            {Object.keys(finalMixObj).map((key) => (
              <div className='d-flex justify-content-between align-items-center'>
                <Card
                  className='p-3 text-center mx-5 my-3'
                  color={
                    selectedWords.some((obj) => obj.key === key)
                      ? 'primary'
                      : ''
                  }
                  onClick={() => {
                    if (selectedWords.some((obj) => obj.key === key)) {
                      handleUnselect({
                        key,
                        value: finalMixObj[key],
                      });
                    } else if (selectedWords.length < 6) {
                      handleSelect({
                        key,
                        value: finalMixObj[key],
                      });
                    }
                  }}
                >
                  <span
                    className={
                      selectedWords.some((obj) => obj.key === key)
                        ? styles.txtWhite
                        : styles.txtPrimary
                    }
                  >
                    {finalMixObj[key]}
                  </span>
                </Card>
              </div>
            ))}
          </Row>
          <div className='text-center'>
            <Button
              color='link'
              size='sm'
              onClick={() => {
                setIsWalletTabActive(!isWalletTabActive);
              }}
            >
              <MdArrowBack />
              <span className='d-md-inline'>
                {I18n.t('containers.wallet.verifyMnemonicPage.showAgain')}
              </span>
            </Button>
          </div>
        </section>
      </div>
      <footer className='footer-bar'>
        {isWalletCreating ? (
          <WalletLoadingFooter
            message={I18n.t(
              'containers.wallet.verifyMnemonicPage.creatingYourWallet'
            )}
          />
        ) : isErrorCreatingWallet.length !== 0 ? (
          <>
            <div className={`footer-sheet`}>
              <div className='text-center'>
                <MdErrorOutline
                  className={classnames({
                    'footer-sheet-icon': true,
                    [styles[`error-dailog`]]: true,
                  })}
                />
                <p>{isErrorCreatingWallet}</p>
              </div>
            </div>
            <div className='d-flex align-items-center justify-content-center'>
              <Button
                color='primary'
                onClick={() => {
                  resetCreateWalletError();
                  history.push(WALLET_PAGE_PATH);
                }}
              >
                {I18n.t(
                  'containers.wallet.verifyMnemonicPage.backToWalletPage'
                )}
              </Button>
            </div>
          </>
        ) : (
          <div>
            <Row className='justify-content-between align-items-center'>
              <Col className='d-flex justify-content-end'>
                <Button
                  color='link'
                  className='mr-3'
                  disabled={!(selectedWords.length === 6) || !mnemonicCheck}
                  onClick={() => {
                    createWallet(mnemonicCode, history);
                  }}
                >
                  {I18n.t('containers.wallet.createNewWalletPage.continue')}
                </Button>
              </Col>
            </Row>
          </div>
        )}
      </footer>
    </>
  );
};

const mapStateToProps = (state) => {
  const { wallet } = state;
  const { isWalletCreating, isErrorCreatingWallet } = wallet;
  return {
    isWalletCreating,
    isErrorCreatingWallet,
  };
};

const mapDispatchToProps = {
  createWallet: (mnemonicCode, history) =>
    createWalletRequest({ mnemonicCode, history }),
  resetCreateWalletError: () => resetCreateWalletError({}),
};

export default connect(mapStateToProps, mapDispatchToProps)(VerifyMnemonic);
