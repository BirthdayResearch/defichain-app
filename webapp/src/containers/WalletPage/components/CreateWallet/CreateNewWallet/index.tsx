import React, { useState } from 'react';
import { Helmet } from 'react-helmet';
import { I18n } from 'react-redux-i18n';
import { NavLink } from 'react-router-dom';
import classnames from 'classnames';
import { MdArrowBack, MdRefresh } from 'react-icons/md';
import {
  Row,
  Col,
  Button,
  FormGroup,
  Label,
  Input,
  Card,
  CardBody,
} from 'reactstrap';

import {
  LEARN_MORE_ABOUT_BITCOIN_LINK,
  WALLET_TOKENS_PATH,
} from '../../../../../constants';
import styles from '../CreateWallet.module.scss';
import Header from '../../../../HeaderComponent';
import openNewTab from '../../../../../utils/openNewTab';
import { getPageTitle } from '../../../../../utils/utility';

interface CreateNewWalletProps {
  mnemonicObj: any;
  isWalletTabActive: boolean;
  generateNewMnemonic: () => void;
  setIsWalletTabActive: (isWalletTabActive: boolean) => void;
}

const CreateNewWallet: React.FunctionComponent<CreateNewWalletProps> = (
  props: CreateNewWalletProps
) => {
  const {
    mnemonicObj,
    generateNewMnemonic,
    isWalletTabActive,
    setIsWalletTabActive,
  } = props;

  const [isChecked, setIsChecked] = useState(false);

  return (
    <>
      <Helmet>
        <title>
          {getPageTitle(I18n.t('containers.wallet.createNewWalletPage.title'))}
        </title>
      </Helmet>
      <Header>
        <Button
          to={WALLET_TOKENS_PATH}
          tag={NavLink}
          color='link'
          className='header-bar-back'
        >
          <MdArrowBack />
          <span className='d-lg-inline'>
            {I18n.t('containers.wallet.createNewWalletPage.back')}
          </span>
        </Button>
        <h1 className={classnames({ 'd-none': false })}>
          {I18n.t('containers.wallet.createNewWalletPage.createANewWallet')}
        </h1>
      </Header>
      <div className='content'>
        <section>
          <p>
            {I18n.t('containers.wallet.createNewWalletPage.mnemonicGuideline')}
          </p>
          <p className={styles.mnemonicGuidelineWarning}>
            {I18n.t(
              'containers.wallet.createNewWalletPage.mnemonicGuidelineWarning'
            )}
          </p>
          <Card className={styles.margin}>
            <CardBody>
              <Row className={styles.seeds}>
                {Object.keys(mnemonicObj).map((key) => (
                  <Col md='4' xs='12' key={key}>
                    <Row>
                      <Col
                        className={`${styles.number} text-right pl-0`}
                        xs='2'
                      >
                        {key}
                      </Col>
                      <Col className='text-left pl-0' xs='10'>
                        {mnemonicObj[key]}
                        <hr />
                      </Col>
                    </Row>
                  </Col>
                ))}
              </Row>
            </CardBody>
          </Card>
          <div className='text-center mt-4'>
            <Button color='link' size='sm' onClick={generateNewMnemonic}>
              <MdRefresh />
              <span className='d-md-inline'>
                {I18n.t('containers.wallet.createNewWalletPage.generateNewSet')}
              </span>
            </Button>
          </div>
          <div className={styles.walletRecoveryInfo}>
            <p>
              {I18n.t(
                'containers.wallet.createNewWalletPage.walletRecoveryInfoLine1'
              )}
            </p>
            <p>
              {I18n.t(
                'containers.wallet.createNewWalletPage.walletRecoveryInfoLine2'
              )}
            </p>
            <Button
              color='link'
              className='p-0'
              onClick={() => openNewTab(LEARN_MORE_ABOUT_BITCOIN_LINK)}
            >
              {I18n.t(
                'containers.wallet.createNewWalletPage.learnMoreAtBitcoinBtn'
              )}
            </Button>
          </div>
        </section>
      </div>
      <footer className='footer-bar'>
        <div>
          <Row className='justify-content-between align-items-center'>
            <Col className='col-auto'>
              <FormGroup check>
                <Label check>
                  <Input
                    type='checkbox'
                    onClick={() => setIsChecked(!isChecked)}
                  />
                  &nbsp;
                  {I18n.t(
                    'containers.wallet.createNewWalletPage.copied24Words'
                  )}
                </Label>
              </FormGroup>
            </Col>
            <Col className='d-flex justify-content-end'>
              <Button
                color='primary'
                disabled={!isChecked}
                onClick={() => {
                  setIsWalletTabActive(!isWalletTabActive);
                }}
              >
                {I18n.t('containers.wallet.createNewWalletPage.continue')}
              </Button>
            </Col>
          </Row>
        </div>
      </footer>
    </>
  );
};

export default CreateNewWallet;
