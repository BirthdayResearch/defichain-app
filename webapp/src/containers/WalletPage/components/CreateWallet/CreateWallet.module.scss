// App and Bootstrap variables required
@import '~bootstrap/scss/functions';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins';
@import '../../../../scss/variables';
@import '../../../../scss/bootstrap/variables';

.statCard {
  margin-bottom: 24px;
  height: calc(100% - 24px);
}

.cardBody {
  display: inline-block;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: flex-end;
}

.label {
  font-weight: $font-weight-medium;
  letter-spacing: 0.15px;
  width: 100%;
  margin-bottom: 0.5rem;
}

.value {
  font-weight: $font-weight-semibold;
  font-size: 2.125rem;
  line-height: 2.5rem;
}

.seeds {
  position: relative;

  &:after {
    content: '';
    position: absolute;
    width: 100%;
    height: 1px;
    background-color: $card-bg;
    bottom: 0;
    left: 0;
  }
}

.seedItem {
  display: flex;
}

.seedNumber {
  text-align: right;
  padding-right: 16px;
  width: 32px;
  flex-shrink: 0;
  white-space: nowrap;
  padding-top: 8px;
  padding-bottom: 8px;
  color: $secondary;
}

.seedWord {
  border-bottom: $border-width solid $border-color;
  flex-grow: 1;
  padding-top: 8px;
  padding-bottom: 8px;
}

.number {
  font-weight: $font-weight-medium;
  color: $secondary;
}

.txtWhite {
  color: #fff;
}

.txtPrimary {
  color: #ff00af;
}

.statCardActionWithIconAnimation {
  svg {
    -webkit-animation: rotation 2s infinite linear;
    animation: rotation 2s infinite linear;
  }
}

.statCardActionWithIcon {
  text-align: end;
}

.error-dialog {
  color: $dfi-red !important;
}

.warning-dailog {
  color: $dfi-orange !important;
}

.mnemonicGuidelineWarning {
  color: $dfi-red;
  margin: 2rem 0;
}

.walletRecoveryInfo {
  background-color: rgba(255, 255, 255, 0);
  color: $secondary;
  border: 1px solid rgba(0, 0, 0, 0.2);
  margin-top: 2rem;
  padding: 2rem;
  border-radius: 8px;
}

.verifyHeadLine6 {
  font-weight: $font-weight-semibold;
  font-size: 20px;
}

.verifySubtitle {
  font-weight: $font-weight-medium;
}

.optionText {
  text-transform: none !important;
}
