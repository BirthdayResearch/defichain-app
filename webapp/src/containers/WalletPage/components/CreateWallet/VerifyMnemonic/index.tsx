import React, { useEffect, useState } from 'react';
import { Helmet } from 'react-helmet';
import { I18n } from 'react-redux-i18n';
import { NavLink } from 'react-router-dom';
import classnames from 'classnames';
import { MdArrowBack, MdErrorOutline, MdWarning } from 'react-icons/md';
import {
  getRandomWordArray,
  selectNfromRange,
} from '../../../../../utils/utility';
import {
  VERIFY_MNEMONIC_QUIZ_OPTIONS_PER_QUESTIONS_LIMIT,
  VERIFY_MNEMONIC_QUIZ_QUESTIONS_LIMIT,
  WALLET_CREATE_PATH,
  WALLET_TOKENS_PATH,
} from '../../../../../constants';
import { Row, Col, Button } from 'reactstrap';

import styles from '../CreateWallet.module.scss';
import { createWalletRequest, resetCreateWalletError } from '../../../reducer';
import { connect } from 'react-redux';
import WalletLoadingFooter from '../../../../../components/WalletLoadingFooter';
import Header from '../../../../HeaderComponent';
import shuffle from 'lodash/shuffle';
import { history } from '../../../../../utils/history';
interface VerifyMnemonicProps {
  mnemonicObj: any;
  finalMixObj: any;
  mnemonicCode: string;
  isWalletTabActive: boolean;
  isWalletCreating: boolean;
  isErrorCreatingWallet: string;
  setIsWalletTabActive: (isWalletTabActive: boolean) => void;
  createWallet: (mnemonicCode: string, history: any) => void;
  resetCreateWalletError: () => void;
}

const VerifyMnemonic: React.FunctionComponent<VerifyMnemonicProps> = (
  props: VerifyMnemonicProps
) => {
  const [quiz, setQuiz] = useState<any[]>([]);
  const [selected, setSelected] = useState<any>({});
  const [enableSubmit, setEnableSubmit] = useState(false);
  const [showSkipVerification, setShowSkipVerification] = useState(false);

  const {
    isWalletTabActive,
    setIsWalletTabActive,
    mnemonicObj,
    isWalletCreating,
    isErrorCreatingWallet,
    mnemonicCode,
    createWallet,
    resetCreateWalletError,
  } = props;

  const handleClick = (wordNum, option) => {
    const clone = Object.assign({}, selected);
    if (!clone[wordNum]) {
      clone[wordNum] = option;
    } else {
      if (clone[wordNum] === option) {
        clone[wordNum] = '';
      } else {
        clone[wordNum] = option;
      }
    }
    setSelected(clone);
  };

  useEffect(() => {
    props.resetCreateWalletError();
    createWalletRecoveryQuiz();
  }, []);

  useEffect(() => {
    quizValidator();
  }, [selected]);

  const quizValidator = () => {
    const selectedKeys = Object.keys(selected).map((item) => !!selected[item]);
    if (selectedKeys.length === 6) {
      const isValidated = quiz.reduce(
        (acc, item) => acc && selected[item.wordNum] === item.correct
      );
      if (isValidated) {
        return setEnableSubmit(true);
      }
    }
    return setEnableSubmit(false);
  };

  const createWalletRecoveryQuiz = () => {
    const lowerBound = 1;
    const upperBound = Object.keys(mnemonicObj).length;
    const limit = VERIFY_MNEMONIC_QUIZ_QUESTIONS_LIMIT;
    const optionsPerQuestions = VERIFY_MNEMONIC_QUIZ_OPTIONS_PER_QUESTIONS_LIMIT;
    const distinctQnum = selectNfromRange(
      lowerBound,
      upperBound,
      limit
    ).sort((a, b) => (a > b ? 1 : -1));
    const randomOptionsWords = getRandomWordArray();
    const updatedData = distinctQnum.map((item, id) => ({
      wordNum: item,
      options: shuffle(
        [mnemonicObj[item]].concat(
          randomOptionsWords.slice(
            id * (optionsPerQuestions - 1),
            (id + 1) * (optionsPerQuestions - 1)
          )
        )
      ),
      correct: mnemonicObj[item],
    }));
    setQuiz(updatedData);
  };
  return (
    <>
      <Helmet>
        <title>{I18n.t('containers.wallet.verifyMnemonicPage.title')}</title>
      </Helmet>
      <Header>
        <Button
          to={WALLET_CREATE_PATH}
          tag={NavLink}
          color='link'
          className='header-bar-back'
        >
          <MdArrowBack />
          <span className='d-lg-inline'>
            {I18n.t('containers.wallet.verifyMnemonicPage.back')}
          </span>
        </Button>
        <h1 className={classnames({ 'd-none': false })}>
          {I18n.t('containers.wallet.verifyMnemonicPage.verifyMnemonicSeed')}
        </h1>
      </Header>
      <div className='content'>
        <section>
          <p>
            {I18n.t(
              'containers.wallet.verifyMnemonicPage.verifyMnemonicGuideline'
            )}
          </p>
          <div className={styles.verifyHeadLine6}>
            {I18n.t(
              'containers.wallet.verifyMnemonicPage.verifyMnemonicWhichOne'
            )}
          </div>
          <Row>
            {quiz.map((item, id) => (
              <Col
                key={item.wordNum + '_' + id}
                xs='12'
                md='6'
                className='my-3'
              >
                <h5>
                  {I18n.t('containers.wallet.verifyMnemonicPage.wordQuestion', {
                    wordNum: item.wordNum,
                  })}
                </h5>
                <Row>
                  <Col xs='12'>
                    {item.options.map((word) => (
                      <span key={item.wordNum + '_' + word} className='mr-3'>
                        <Button
                          color={
                            word === selected[item.wordNum]
                              ? 'primary'
                              : 'outline-primary'
                          }
                          className={styles.optionText}
                          onClick={() => handleClick(item.wordNum, word)}
                        >
                          {word}
                        </Button>
                      </span>
                    ))}
                  </Col>
                </Row>
              </Col>
            ))}
          </Row>
        </section>
      </div>
      <footer className='footer-bar'>
        {isWalletCreating ? (
          <WalletLoadingFooter
            message={I18n.t(
              'containers.wallet.verifyMnemonicPage.creatingYourWallet'
            )}
          />
        ) : isErrorCreatingWallet.length !== 0 ? (
          <>
            <div className={`footer-sheet`}>
              <div className='text-center'>
                <MdErrorOutline
                  className={classnames({
                    'footer-sheet-icon': true,
                    [styles[`error-dialog`]]: true,
                  })}
                />
                <p>{isErrorCreatingWallet}</p>
              </div>
            </div>
            <div className='d-flex align-items-center justify-content-center'>
              <Button
                color='primary'
                onClick={() => {
                  resetCreateWalletError();
                  history.push(WALLET_TOKENS_PATH);
                }}
              >
                {I18n.t(
                  'containers.wallet.verifyMnemonicPage.backToWalletPage'
                )}
              </Button>
            </div>
          </>
        ) : (
          <div>
            {showSkipVerification && (
              <div className={`footer-sheet`}>
                <div className='text-center'>
                  <MdWarning
                    className={classnames({
                      'footer-sheet-icon': true,
                      [styles[`warning-dailog`]]: true,
                    })}
                  />
                  <h3>
                    {I18n.t(
                      'containers.wallet.verifyMnemonicPage.skipVerificationTextHeader'
                    )}
                  </h3>
                  <p>
                    {I18n.t(
                      'containers.wallet.verifyMnemonicPage.skipVerificationText'
                    )}
                  </p>
                </div>
              </div>
            )}
            <Row className='justify-content-between align-items-center'>
              <Col>
                {!showSkipVerification && (
                  <Button
                    color='link'
                    size='sm'
                    className='footer-bar-back'
                    onClick={() => {
                      setIsWalletTabActive(!isWalletTabActive);
                    }}
                  >
                    <MdArrowBack />
                    <span className='d-md-inline'>
                      {I18n.t('containers.wallet.verifyMnemonicPage.showAgain')}
                    </span>
                  </Button>
                )}
              </Col>
              <Col className='d-flex justify-content-end'>
                {/* <Button
                  color='link'
                  onClick={() =>
                    setShowSkipVerification(!showSkipVerification)
                  }
                >
                  {showSkipVerification
                    ? I18n.t('containers.wallet.verifyMnemonicPage.dontSkip')
                    : I18n.t('containers.wallet.verifyMnemonicPage.skip')}
                </Button> */}
                <Button
                  color={showSkipVerification ? 'primary' : 'primary'}
                  disabled={!(showSkipVerification || enableSubmit)}
                  onClick={() => createWallet(mnemonicCode, history)}
                >
                  {showSkipVerification
                    ? I18n.t('containers.wallet.verifyMnemonicPage.skip')
                    : I18n.t('containers.wallet.verifyMnemonicPage.continue')}
                </Button>
              </Col>
            </Row>
          </div>
        )}
      </footer>
    </>
  );
};

const mapStateToProps = (state) => {
  const { wallet } = state;
  const { isWalletCreating, isErrorCreatingWallet } = wallet;
  return {
    isWalletCreating,
    isErrorCreatingWallet,
  };
};

const mapDispatchToProps = {
  createWallet: (mnemonicCode, history) =>
    createWalletRequest({ mnemonicCode, history }),
  resetCreateWalletError: () => resetCreateWalletError({}),
};

export default connect(mapStateToProps, mapDispatchToProps)(VerifyMnemonic);
