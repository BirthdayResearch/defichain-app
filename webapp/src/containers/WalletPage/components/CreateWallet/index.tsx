import { connect } from 'react-redux';
import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { createWalletStart, createWalletFailure } from '../../reducer';
import EncryptWalletPage, { EncryptWalletPayload } from '../EncryptWalletPage';

interface CreateWalletProps extends RouteComponentProps {
  createWalletStart: (item: EncryptWalletPayload) => void;
  createWalletFailure: (err: string) => void;
  isWalletCreating: boolean;
  isErrorCreatingWallet: string;
}

const CreateWallet: React.FunctionComponent<CreateWalletProps> = (
  props: CreateWalletProps
) => {
  const {
    createWalletStart,
    createWalletFailure,
    isWalletCreating,
    isErrorCreatingWallet,
  } = props;
  return (
    <EncryptWalletPage
      submitButtonLabel={
        'containers.wallet.createNewWalletPage.createANewWallet'
      }
      onSave={createWalletStart}
      pageLoadingMessage={
        'containers.wallet.verifyMnemonicPage.creatingYourWallet'
      }
      isPageLoading={isWalletCreating}
      pageErrorMessage={isErrorCreatingWallet}
      onCloseFailure={createWalletFailure}
      pageSize={11}
    />
  );
};

const mapStateToProps = (state) => {
  const { isWalletCreating, isErrorCreatingWallet } = state.wallet;
  return {
    isWalletCreating,
    isErrorCreatingWallet,
  };
};

const mapDispatchToProps = {
  createWalletStart,
  createWalletFailure,
};

export default connect(mapStateToProps, mapDispatchToProps)(CreateWallet);
