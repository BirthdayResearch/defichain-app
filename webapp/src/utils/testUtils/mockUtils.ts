import PersistentStore from '../persistentStore';
import * as isElectron from '../isElectron';
import * as Notifications from '../notifications';
import log from 'loglevel';
import { runSaga } from 'redux-saga';
import axios from 'axios';

const mockPersistentStore = (getData, setData) => {
  PersistentStore.get = jest.fn(() => getData);
  PersistentStore.set = jest.fn(() => setData);
  return PersistentStore;
};

const mockAxios = mockFunc => {
  axios.create = jest.fn().mockImplementation(() => ({
    post: mockFunc,
  }));
};
const dispatchedFunc = async (method, payload = {}) => {
  const dispatched = Array();
  let action = {};
  if (payload) action = { payload };
  await runSaga(
    {
      dispatch: action => dispatched.push(action),
    },
    method,
    action
  );
  return dispatched;
};

const mockisElectron = () => {
  return jest.spyOn(isElectron, 'isElectron');
};
const mockIpcRenderer = () => {
  return jest.spyOn(isElectron, 'ipcRendererFunc');
};

const spyLogger = () => {
  return jest.spyOn(log, 'error');
};

const mockNotification = () => {
  return (
    jest
      .spyOn(Notifications, 'default')
      // tslint:disable-next-line: no-empty
      .mockImplementation(() => {})
  );
};
export {
  mockPersistentStore,
  mockAxios,
  dispatchedFunc,
  mockisElectron,
  mockIpcRenderer,
  spyLogger,
  mockNotification,
};
