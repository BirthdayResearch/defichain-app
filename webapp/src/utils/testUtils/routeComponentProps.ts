import { createMemoryHistory, createLocation } from 'history';
import { match } from 'react-router';
const history = createMemoryHistory();
const path = `/route/:id`;
const match: match<{ id: string }> = {
  isExact: false,
  path,
  url: path.replace(':id', '1'),
  params: { id: '1' },
};

const location = createLocation(match.url);
delete history.location.key;
delete location.key;

export { history, match, location };
