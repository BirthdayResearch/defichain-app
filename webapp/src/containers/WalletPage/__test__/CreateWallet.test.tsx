import React from 'react';
import CreateWallet from '../components/CreateWallet';
import * as routeComponentProps from '../../../utils/testUtils/routeComponentProps';
import { shallow } from 'enzyme';

describe('CreateWallet component', () => {
  it('should check for snapshot', () => {
    const wrapper = shallow(<CreateWallet {...routeComponentProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
