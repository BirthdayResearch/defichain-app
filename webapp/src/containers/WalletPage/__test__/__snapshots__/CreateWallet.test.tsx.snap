// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateWallet component should check for snapshot 1`] = `
<div
  className="main-wrapper"
>
  <CreateNewWallet
    generateNewMnemonic={[Function]}
    isWalletTabActive={false}
    mnemonicObj={Object {}}
    setIsWalletTabActive={[Function]}
  />
</div>
`;
